youtube_transcript_api/formatters.py

```diff
@@ -0,0 +1,174 @@
+from collections import defaultdict
+import json
+import re
+
+
+def parse_timecode(time):
+    """Converts a `time` into a formatted transcript timecode.
+
+    :param time: a float representing time in seconds.
+    :type time: float
+    :return: a string formatted as a timecode, 'HH:MM:SS,MS'
+    :rtype str
+
+    :example:
+    >>> parse_timecode(6.93)
+    '00:00:06,930'
+    """
+    
+    time = float(time)
+    times = {
+        'hours': str(int(time) // 3600).rjust(2, '0'),
+        'mins': str(int(time) // 60).rjust(2, '0'),
+        'secs': str(int(time) % 60).rjust(2, '0'),
+        'ms': str(int(round((time - int(time))*1000, 2))).rjust(3, '0')
+    }
+    return "{hours}:{mins}:{secs},{ms}".format(**times)
+
+
+class TranscriptFormatter(object):
+    """Abstract Base TranscriptFormatter class
+
+    This class should be inherited from to create additional
+     custom transcript formatters.
+    """
+    HTML_TAG_REGEX = re.compile(r'<[^>]*>', re.IGNORECASE)
```

Comment:
Ahh, you might be right. This was likely a left-over artifact from my initial idea I had for implementation. 

So my base class was originally this idea of a "parser" for changing the formatting instead of formatter. So that I could parse the incoming "plain data" before it was ever converted to its List<Dict> format by **parser code you have in place below**. So we could skip that initial processing and allow the user to customize that data earlier on.

I got this working, but would of have to overhaul at least all of the current tests you had. But it helped going from that right into making it "Formatters" instead. 


```python
class _TranscriptParser():
    HTML_TAG_REGEX = re.compile(r'<[^>]*>', re.IGNORECASE)

    def parse(self, plain_data):
        # ... code... 
```